classes/cron_processor.php

```diff
@@ -93,6 +96,14 @@ public function get_min_duration(): float {
      * @param manager $manager
      */
     public function on_interval(manager $manager) {
+        // We want to prevent doubling up of processing, so skip if an existing process is still executing.
+        // The profile logs will be kept and processed the next time.
+        if (self::$isinsideoninterval) {
```

Comment:
in cases when this has happened sometimes this is a very long running script, I would not want to have say 100s of the same debugging emitted in these cases I think just one call would be sufficient

Reply:
fixed

Reply:
I also changed it so that it suspends future partial saves, as suggested in the WR.